generator/validate/gen.go

```diff
@@ -0,0 +1,172 @@
+package validate
```

Comment:
Should we have license header for this?

Reply:
https://github.com/devfile/api/pull/137 :)